### ISSUE TITLE
Override that narrows a receiver's type arguments is not reported

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2076]
The Nullness Checker accepts an override whose receiver requires a stricter type argument than the overridden method's receiver. This lets a call through the parent type throw a `NullPointerException`.

Found during my Lean formalization and by tests generated by a program enumerator.

```java
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

class Parent<E extends @Nullable Object> {
    E value;

    Parent(E value) {
        this.value = value;
    }

    void inspect(Parent<E> this) {}
}

class Child<E extends @Nullable Object> extends Parent<E> {
    Child(E value) {
        super(value);
    }

    @Override
    void inspect(Child<@NonNull E> this) {
        value.toString();
    }
}

class Use {
    public static void main(String[] args) {
        Child<@Nullable String> child = new Child<>(null);
        Parent<@Nullable String> parent = child;
        parent.inspect();
    }
}
```

Command: `javac -processor nullness ReceiverArgOverride.java`

**Observed:** no errors. Running the program throws:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "Object.toString()" because "this.value" is null
	at Child.inspect(ReceiverArgOverride.java:22)
	at Use.main(...)
```

**Expected:** an `override.receiver.invalid` error on `Child.inspect`.

For comparison, calling `child.inspect()` directly is correctly rejected with `method.invocation.invalid` (found `Child<@Nullable String>`, required `Child<@NonNull String>`).

**Environment:** EISOP Checker Framework at 864d54458c, JDK 21, macOS.
